R/layers.R

```diff
@@ -943,6 +943,53 @@ addGeoJSON = function(map, geojson, layerId = NULL, group = NULL,
   invokeMethod(map, getMapData(map), 'addGeoJSON', geojson, layerId, group, options)
 }
 
+#' @param style a JSON string
+#' @describeIn map-layers change style of GeoJSON feature
+#' @export
+styleFeatureGeoJSON = function(map, layerId = NULL, featureId = NULL, style = NULL) {
+  if(class(layerId) != "character" || length(layerId) != 1) {
+    stop("Missing layerId string")
+  }
+  if(class(featureId) != "character" || length(featureId) != 1) {
+    stop("Missing featureId string")
+  }
+  if(class(style)=="list") {
+    styleJSON <- as.character(jsonlite::toJSON(style))
```

Comment:
Why manually encode the list here and `JSON.parse()` later in JS? This seems to be unnecessary to me. I'd remove the check of `class(style)` here, accept whatever the user inputs, then in JS use `JSON.parse()` only if necessary (cf `methods.addGeoJSON()`).